### DISCUSSION TITLE
Can polls be enabled for not logged in visitors?

### DISCUSSION THREAD

**Discussion by canuckdev** [mastodon/mastodon#17458] [Q&A]
A poll displayed on page https://mastodon_domain/@user is disabled whether logged in or not. The poll can only be voted on by a logged in user from the poll displayed on https://mastodon_domain/web/timelines/*.

Is it possible to make poll voting public? The following code seems to have something to do with this, but I'm not quite sure what I need to change. Any advice will be much appreciated!

```
# mastodon-3.4.6/app/controllers/api/v1/polls_controller.rb

# frozen_string_literal: true

class Api::V1::PollsController < Api::BaseController
  include Authorization

  before_action -> { authorize_if_got_token! :read, :'read:statuses' }, only: :show
  before_action :set_poll
  before_action :refresh_poll

  def show
    render json: @poll, serializer: REST::PollSerializer, include_results: true
  end

  private

  def set_poll
    @poll = Poll.attached.find(params[:id])
    authorize @poll.status, :show?
  rescue Mastodon::NotPermittedError
    not_found
  end

  def refresh_poll
    ActivityPub::FetchRemotePollService.new.call(@poll, current_account) if user_signed_in? && @poll.possibly_stale?
  end
end
```

**Comment by ryanmio** [comment]
To allow non-logged-in users to vote, you'd have to modify the authorization logic in PollsController and potentially other parts of the system

`before_action -> { authorize_if_got_token! :read, :'read:statuses' }, only: :show
`
